### ISSUE TITLE
Allow host names (with wild cards) in Search Preference 

### ISSUE THREAD

**Issue by ppat** [elastic/elasticsearch#5925]
I have a single index (multiple shards) that is pinned to two groups nodes using shard allocation filtering.

```
curl -XPUT localhost:9200/test/_settings -d '{
    "index.routing.allocation.include._host" : "es-large-*"
    "index.routing.allocation.include._host" : "es-small-*"
}'
```

There are 16 large nodes and 4 small nodes. I have allocated 17 replicas (for a total of 18) so at least 2 of the 4 small nodes will have a copy of every shard.

I have two processes querying this same index. One of them is powering a real-time user experience and is low throughput (low query rate) and low latency queries. The other process is powering something near-real-time and executes high latency queries with high throughput (high query rate).

I want to target the small nodes for the low latency, low throughput queries and the large nodes for the high latency, high throughput queries. That way any performance impact from the high latency queries will not affect the real time user experience powered by low low latency queries. 

Is it possible to update Search Preference so that I can target the large and small nodes accordingly?

**Comment by dakrone** [comment]
@ppat I think this is the same issue as #10229, which can be solved by using [shard allocation awareness](http://www.elastic.co/guide/en/elasticsearch/reference/current/modules-cluster.html#_automatic_preference_when_searching_geting), in that case, ES will automatically try to limit search responses within a particular awareness attribute.

Hopefully this solves this issue, let me know if we correctly understood your issue.

**Comment by ppat** [comment]
It's not the same issue. We're already  using shard allocation awareness to ensure that nodes do not end up on the same VM host machine. We also use shard allocation filtering to pin indices to nodes. 

This request is adding another dimension at search time to target only hosts that match a specific regex. Since I created this issue about 12 months ago, I have found a work around at the cost of latency. I basically run a search shards request to figure which nodes the shards I need to query are on and then determine which one of them matches my regex for host name pattern (different regex pattern for different applications). Then use matching node-id with search preference when sending the actual search query to ES.

**Comment by clintongormley** [comment]
I like this idea.  Would be good to support the same node selectors as we have here: http://www.elastic.co/guide/en/elasticsearch/reference/current/cluster.html

**Comment by ppat** [comment]
Multiple ways of being able to specify node names would indeed be great. In that link, it's not clear whether host names are supported (hopefully they are... Including use of wildcards with host name).

**Comment by ppf2** [comment]
+1

**Comment by nirmalc** [comment]
+1 , also on node attribs 

_only_node_attribs:<food>

**Comment by clintongormley** [comment]
closed by #11464
